src/main/cpp/src/decimal_utils.cu

```diff
@@ -520,6 +519,67 @@ inline __device__ int precision10(chunked256 value) {
     return -1;
 }
 
+__device__ chunked256 set_scale_and_round(chunked256 data, int old_scale, int new_scale) {
+  if (old_scale != new_scale) {
+    if (new_scale < old_scale) {
+      int raise = old_scale - new_scale;
```

Comment:
Use `const` for immutable variables as much as possible. Not just here but everywhere else if possible.

Reply:
My rule of thumb is to define as `auto const`, and then remove the const when the need to modify presents itself, not unlike default `val` in Scala.